config/src/config/config_optimizer.rs

```diff
@@ -4,17 +4,47 @@
 use crate::{
     config::{
         node_config_loader::NodeType, utils::get_config_name, Error, InspectionServiceConfig,
-        LoggerConfig, MempoolConfig, NodeConfig, PeerMonitoringServiceConfig, StateSyncConfig,
+        LoggerConfig, MempoolConfig, NodeConfig, Peer, PeerMonitoringServiceConfig, PeerRole,
+        PeerSet, StateSyncConfig,
     },
     network_id::NetworkId,
 };
-use aptos_types::chain_id::ChainId;
+use aptos_crypto::{x25519, ValidCryptoMaterialStringExt};
+use aptos_types::{chain_id::ChainId, network_address::NetworkAddress, PeerId};
+use maplit::hashset;
 use serde_yaml::Value;
+use std::{collections::HashMap, str::FromStr};
 
 // Useful optimizer constants
 const OPTIMIZER_STRING: &str = "Optimizer";
+const PUBLIC_NETWORK_OPTIMIZER_NAME: &str = "PublicNetworkConfigOptimizer";
 const VALIDATOR_NETWORK_OPTIMIZER_NAME: &str = "ValidatorNetworkConfigOptimizer";
 
+// Mainnet seed peer addresses and keys
+const MAINNET_SEED_PEER_ADDRESSES: [&str; 0] = [];
+const MAINNET_SEED_PEER_NETWORK_ADDRESSES: [&str; 0] = [];
+const MAINNET_SEED_PEER_PUBLIC_KEYS: [&str; 0] = [];
+
+// Testnet seed peer addresses and keys
+const TESTNET_SEED_PEER_ADDRESSES: [&str; 4] = [
```

Comment:
I wish there was either a loud comment "these three are one thing and must be the same length and correspond by index", or (really this) I wish it was one list of something with three elements. maybe [[&str; 3]; 4] or vec![] of (str, str, str) tuples?

Reply:
SGTM! I've gone for an array of tuples of (str, str, str) 😄